compiler/frontend/src/org/jetbrains/kotlin/parsing/KotlinParsing.java

```diff
@@ -112,9 +112,20 @@ void parseFile() {
             parseTopLevelDeclaration();
         }
 
+        checkUnclosedBlockComment();
         fileMarker.done(KT_FILE);
     }
 
+    private void checkUnclosedBlockComment() {
+        if (TokenSet.create(BLOCK_COMMENT, DOC_COMMENT).contains(myBuilder.rawLookup(-1))) {
+            int startOffset = myBuilder.rawTokenTypeStart(-1);
+            int endOffset = myBuilder.rawTokenTypeStart(0);
+            if (!myBuilder.getOriginalText().subSequence(startOffset, endOffset).toString().endsWith("*/")) {
+                error("Unclosed comment");
```

Comment:
Thank you! I have used your idea. Java doesn't have check for unclosed comments in parser at all. It is performed in `HighlightUtil` instead.